src/components/Todo/TodoHeader.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  // prettier-ignore
+  useEffect,
+  useRef,
+  useContext,
+  FormEvent,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import { useTodoContext } from '../../store/todoContext';
+import { AuthContext } from '../Auth/AuthContext';
+import { postTodos } from '../../api/todos';
+import { ErrorMessage } from '../../types/Error';
+
+/* eslint-disable jsx-a11y/control-has-associated-label */
+export const TodoHeader = () => {
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const [isAdding, setIsAdding] = useState(false);
+
+  const {
+    // prettier-ignore
+    newTodo,
+    getNewTodo,
+    renewTodos,
+    setError,
+    addTempTodo,
+    isAllTodosCompleted,
+    toggleAllTodos,
+  } = useTodoContext();
+
+  const user = useContext(AuthContext);
+
+  const sendTodo = async (e: FormEvent) => {
+    e.preventDefault();
+    if (newTodo && user) {
+      const todoToSend = {
+        userId: user.id,
+        title: newTodo,
+        completed: false,
+      };
+
+      addTempTodo(newTodo, user.id);
+
+      try {
+        setIsAdding(true);
+
+        const todo = await postTodos(todoToSend);
+        // eslint-disable-next-line
+        renewTodos(todo);
+      } catch {
+        setError(true, ErrorMessage.AddError);
+      } finally {
+        getNewTodo('');
+        addTempTodo();
+        setIsAdding(false);
+      }
+    } else {
+      setError(true, ErrorMessage.TitleError);
+    }
+  };
+
+  const toggleAll = () => {
+    if (!isAllTodosCompleted) {
+      return;
+    }
+
+    toggleAllTodos();
+  };
+
+  useEffect(() => {
+    // focus the element with `ref={newTodoField}`
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, []);
```

Comment:
this doesn't work - after you submit new todo, input field is not focused

Reply:
deps array is empty - hence it happens only once - on componentDidMount
so you have to add a dependency which will trigger the .focus on it's change